### ISSUE TITLE
List all the metadata returned by neural networks benchmarks

### ISSUE THREAD

**Issue by Deathn0t** [deephyper/benchmark#4]
The list of metadata returned by all benchmarks tuning neural networks should be listed a current list is:

- `num_parameters`: integer value of the number of trainable parameters of the neural network.
- `budget`: scalar value (float/int) of the budget consumed by the neural network. Therefore the budget should be defined for each benchmark (e.g., number of epochs in general).
- `stopped`: boolean value indicating if the evaluation was stopped before consuming the maximum budget.
- `train_X`:  scalar value of the training metrics (replace `X` by the metric name, 1 key per metric).
- `valid_X`: scalar value of the validation metrics (replace `X` by the metric name, 1 key per metric).
- `test_X`: scalar value of the testing metrics (replace `X` by the metric name, 1 key per metric).

Also add the `@profile` decorator to all functions.

**Comment by Deathn0t** [comment]
Added to the README in https://github.com/deephyper/benchmark/commit/1d2c7211bcff5ecb29b1710c68b844bb7ed1144d
